validate/josev2/validator.go

```diff
@@ -14,21 +14,11 @@ import (
 type Validator struct {
 	keyFunc            func(context.Context) (interface{}, error) // Required.
 	signatureAlgorithm jose.SignatureAlgorithm                    // Required.
```

Comment:
Can we do this without depending on `jose.SignatureAlgorithm`, so we could swap it out in the future if needed, e.g. requiring a `string` instead?

Reply:
Or providing our own set of valid constants?